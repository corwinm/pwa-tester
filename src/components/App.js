import { AppStatusProvider } from "context/AppStatusContext";
import { BrowserRouter as Router, Route } from "react-router-dom";
import AppUpdate from "components/AppUpdate";
import ErrorBoundary from "components/ErrorBoundary";
import Footer from "components/Footer";
import Home from "components/Home";
import Navbar from "components/Navbar";
import OfflineIndicator from "components/OfflineIndicator";
import React, { Suspense, lazy } from "react";
import RouteTrace from "components/RouteTrace";

const LazyAbout = lazy(() => import("components/About"));
const LazyCamera = lazy(() => import("components/Camera"));
const LazyDeviceMotion = lazy(() => import("components/DeviceMotion"));
const LazyOrientation = lazy(() => import("components/DeviceOrientation"));
const LazyGeolocation = lazy(() => import("components/Geolocation"));
const LazyNotifications = lazy(() => import("components/Notifications"));

const App = () => {
  return (
    <div className="App">
      <AppStatusProvider>
        <Router basename={process.env.PUBLIC_URL}>
          <RouteTrace />
          <Navbar />
          <OfflineIndicator />
          <main role="main">
            <ErrorBoundary>
              <Suspense fallback={<div>Loading...</div>}>
                <Route exact path="/" component={Home} />
                <Route path="/about" component={LazyAbout} />
                <Route path="/geolocation" component={LazyGeolocation} />
                <Route path="/notifications" component={LazyNotifications} />
                <Route path="/camera" component={LazyCamera} />
                <Route path="/device/orientation" component={LazyOrientation} />
                <Route path="/device/motion" component={LazyDeviceMotion} />
              </Suspense>
            </ErrorBoundary>
          </main>
          <Footer />
          <AppUpdate />
        </Router>
      </AppStatusProvider>
    </div>
  );
};

export default App;
