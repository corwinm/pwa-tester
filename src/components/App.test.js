import React from "react";
import ReactDOM from "react-dom";
import { render, cleanup } from "@testing-library/react";
import App from "./App";

jest.mock("react-ga", () => {
  return {
    initialize: jest.fn(),
    pageview: jest.fn(),
  };
});

jest.mock("components/BuildTimeStamp", () => () => "Mock Timestamp");

afterEach(cleanup);

test("renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});

test("matches snapshot", () => {
  const { container } = render(<App />);
  expect(container.firstChild).toMatchSnapshot();
});
