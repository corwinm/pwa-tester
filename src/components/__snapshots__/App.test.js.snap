// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  class="App"
>
  <nav
    aria-label="main navigation"
    class="navbar is-transparent is-fixed-top notch-inset-item"
    role="navigation"
  >
    <div
      class="navbar-brand"
    >
      <a
        aria-current="page"
        class="navbar-item active"
        href="/"
      >
        PWA Tester
      </a>
      <a
        aria-expanded="false"
        aria-label="menu"
        class="navbar-burger burger "
        role="button"
        tabindex="0"
      >
        <span
          aria-hidden="true"
        />
        <span
          aria-hidden="true"
        />
        <span
          aria-hidden="true"
        />
      </a>
    </div>
    <div
      class="navbar-menu "
    >
      <div
        class="navbar-start"
      >
        <a
          aria-current="page"
          class="navbar-item has-text-link"
          href="/"
        >
          Home
        </a>
        <a
          class="navbar-item"
          href="/about"
        >
          About
        </a>
        <a
          class="navbar-item"
          href="/geolocation"
        >
          Geolocation
        </a>
        <a
          class="navbar-item"
          href="/notifications"
        >
          Notifications
        </a>
        <a
          class="navbar-item"
          href="/camera"
        >
          Camera
        </a>
        <div
          class="navbar-item has-dropdown is-hoverable"
        >
          <a
            class="navbar-link"
            tabindex="0"
          >
            Device
          </a>
          <div
            class="navbar-dropdown is-boxed"
          >
            <a
              class="navbar-item"
              href="/device/orientation"
            >
              Orientation
            </a>
            <a
              class="navbar-item"
              href="/device/motion"
            >
              Motion
            </a>
          </div>
        </div>
      </div>
      <div
        class="navbar-end"
      >
        <div
          class="navbar-item"
        >
          <div
            class="buttons"
          >
            <button
              class="button is-primary"
              disabled=""
            >
              <span
                class="icon"
              >
                <i
                  class="fas fa-share"
                />
              </span>
              <span>
                Share
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </nav>
  <main
    role="main"
  >
    <section
      class="hero is-primary notch-inset-item"
    >
      <div
        class="hero-body"
      >
        <div
          class="container"
        >
          <h1
            class="title"
          >
            ServiceWorker NOT Supported!
          </h1>
          <button
            class="button is-primary is-inverted is-outlined"
          >
            Call to Action!
          </button>
        </div>
      </div>
    </section>
    <div
      class="notch-inset-item"
    >
      <section
        class="section"
      >
        <div
          class="container is-fluid"
        >
          <h1
            class="title has-text-white"
          >
            Features
          </h1>
          <div>
            <div
              class="tile is-ancestor"
            >
              <div
                class="tile is-parent"
              >
                <article
                  class="tile is-child box notification is-danger has-text-black"
                >
                  <p
                    class="title"
                  >
                    Geolocation
                  </p>
                  <p
                    class="subtitle"
                  >
                    Not Supported
                  </p>
                </article>
              </div>
              <div
                class="tile is-parent"
              >
                <article
                  class="tile is-child box notification is-danger has-text-black"
                >
                  <p
                    class="title"
                  >
                    Camera
                  </p>
                  <p
                    class="subtitle"
                  >
                    Not Supported
                  </p>
                </article>
              </div>
              <div
                class="tile is-parent"
              >
                <article
                  class="tile is-child box notification is-danger has-text-black"
                >
                  <p
                    class="title"
                  >
                    Notifications
                  </p>
                  <p
                    class="subtitle"
                  >
                    Not Supported
                  </p>
                </article>
              </div>
            </div>
            <div
              class="tile is-ancestor"
            >
              <div
                class="tile is-parent"
              >
                <article
                  class="tile is-child box notification is-danger has-text-black"
                >
                  <p
                    class="title"
                  >
                    Payments
                  </p>
                  <p
                    class="subtitle"
                  >
                    Not Supported
                  </p>
                </article>
              </div>
              <div
                class="tile is-parent"
              >
                <article
                  class="tile is-child box notification is-danger has-text-black"
                >
                  <p
                    class="title"
                  >
                    Motion
                  </p>
                  <p
                    class="subtitle"
                  >
                    Not Supported
                  </p>
                </article>
              </div>
              <div
                class="tile is-parent"
              >
                <article
                  class="tile is-child box notification is-danger has-text-black"
                >
                  <p
                    class="title"
                  >
                    Orientation
                  </p>
                  <p
                    class="subtitle"
                  >
                    Not Supported
                  </p>
                </article>
              </div>
            </div>
            <div
              class="tile is-ancestor"
            >
              <div
                class="tile is-parent"
              >
                <article
                  class="tile is-child box notification is-danger has-text-black"
                >
                  <p
                    class="title"
                  >
                    Vibration
                  </p>
                  <p
                    class="subtitle"
                  >
                    Not Supported
                  </p>
                </article>
              </div>
              <div
                class="tile is-parent"
              >
                <article
                  class="tile is-child box notification is-success has-text-black"
                >
                  <p
                    class="title"
                  >
                    Online Status
                  </p>
                  <p
                    class="subtitle"
                  >
                    Supported
                     
                  </p>
                </article>
              </div>
              <div
                class="tile is-parent"
              >
                <article
                  class="tile is-child box notification is-danger has-text-black"
                >
                  <p
                    class="title"
                  >
                    Clipboard
                  </p>
                  <p
                    class="subtitle"
                  >
                    Not Supported
                  </p>
                </article>
              </div>
            </div>
            <div
              class="tile is-ancestor"
            >
              <div
                class="tile is-parent"
              >
                <article
                  class="tile is-child box notification is-danger has-text-black"
                >
                  <p
                    class="title"
                  >
                    Web Share
                  </p>
                  <p
                    class="subtitle"
                  >
                    Not Supported
                  </p>
                </article>
              </div>
              <div
                class="tile is-parent"
              >
                <article
                  class="tile is-child box notification is-danger has-text-black"
                >
                  <p
                    class="title"
                  >
                    Push Notifications
                  </p>
                  <p
                    class="subtitle"
                  >
                    Not Supported
                  </p>
                </article>
              </div>
              <div
                class="tile is-parent"
              >
                <article
                  class="tile is-child box notification is-danger has-text-black"
                >
                  <p
                    class="title"
                  >
                    Background Sync
                  </p>
                  <p
                    class="subtitle"
                  >
                    Not Supported
                  </p>
                </article>
              </div>
            </div>
          </div>
        </div>
      </section>
    </div>
  </main>
  <footer
    class="footer"
  >
    <div
      class="content has-text-centered"
    >
      <p>
        PWA Tester by 
        <a
          href="https://github.com/corwinm"
        >
          Corwin Marsh
        </a>
        . The source code is available on
         
        <a
          href="https://github.com/corwinm/pwa-tester"
        >
           
          <span
            class="icon"
          >
            <i
              class="fab fa-github"
            />
          </span>
          GitHub
        </a>
        <a
          href="https://googlechrome.github.io/lighthouse/viewer/?gist=6ac7f4fd8b349c903dfcf04d2ff88e3e"
        >
           
          <span
            class="icon"
          >
            <i
              class="fab fa-chrome"
            />
          </span>
          Lighthouse Score
        </a>
      </p>
      Mock Timestamp
    </div>
  </footer>
</div>
`;
